app/models/manageiq/providers/openstack/cloud_manager.rb

```diff
@@ -419,52 +427,19 @@ def ensure_managers
     ensure_network_manager
     ensure_cinder_manager
     ensure_swift_manager
+    # TODO: remove when this moves to before_initialization
```

Comment:
Do you mean before_validation?

Reply:
I think we should create the objects at initialization time - empty vessels
then in before validation, make sure they have the correct values

I seem to remember that rails could auto build the `has_one`  for us but can't find the references anymore.

But maybe before validation is where we will create these objects too